dll/win32/shell32/CDefView.cpp

```diff
@@ -1587,6 +1587,26 @@ LRESULT CDefView::OnExplorerCommand(UINT uCommand, BOOL bUseSelection)
     if (FAILED_UNEXPECTEDLY( hResult))
         goto cleanup;
 
+    if (bUseSelection)
+    {
+        // FIXME: we should cache this....
+        SFGAOF rfg = SFGAO_BROWSABLE | SFGAO_CANCOPY | SFGAO_CANLINK | SFGAO_CANMOVE | SFGAO_CANDELETE | SFGAO_CANRENAME | SFGAO_HASPROPSHEET | SFGAO_FILESYSTEM | SFGAO_FOLDER;
+        hResult = m_pSFParent->GetAttributesOf(m_cidl, m_apidl, &rfg);
+        if (FAILED_UNEXPECTEDLY(hResult))
+            return 0;
+
+        if (!(rfg & SFGAO_CANMOVE) && uCommand == FCIDM_SHVIEW_CUT)
+            return 0;
+        if (!(rfg & SFGAO_CANCOPY) && uCommand == FCIDM_SHVIEW_COPY)
+            return 0;
+        if (!(rfg & SFGAO_CANDELETE) && uCommand == FCIDM_SHVIEW_DELETE)
+            return 0;
+        if (!(rfg & SFGAO_CANRENAME) && uCommand == FCIDM_SHVIEW_RENAME)
+            return 0;
+        if (!(rfg & SFGAO_HASPROPSHEET) && uCommand == FCIDM_SHVIEW_PROPERTIES)
+            return 0;
```

Comment:
```suggestion
        if (!(rfg & SFGAO_CANMOVE) && uCommand == FCIDM_SHVIEW_CUT ||
            !(rfg & SFGAO_CANCOPY) && uCommand == FCIDM_SHVIEW_COPY ||
            !(rfg & SFGAO_CANDELETE) && uCommand == FCIDM_SHVIEW_DELETE ||
            !(rfg & SFGAO_CANRENAME) && uCommand == FCIDM_SHVIEW_RENAME ||
            !(rfg & SFGAO_HASPROPSHEET) && uCommand == FCIDM_SHVIEW_PROPERTIES)
            return 0;
```

Reply:
Why?

Reply:
It simplifies the code and avoids using multiple returns, when it's possible to use only one.
Sorry if this appears to be too trivial.

Reply:
The compiler should optimize this anyway, better readable code is better.

Reply:
OK. You know better.